### ISSUE TITLE
change doc generation to go tooling based comments

### ISSUE THREAD

**Issue by butonic** [owncloud/ocis#9290]
The current go tags in our configuration only allok single line comments and make the config file hard to parse / understand for humans, eg:
```go
	Name string `yaml:"name" env:"APP_PROVIDER_SERVICE_NAME" desc:"The name of the service. This needs to be changed when using more than one app provider. Each app provider configured needs to be identified by a unique service name. Possible examples are: 'app-provider-collabora', 'app-provider-onlyoffice', 'app-provider-office365'." introductionVersion:"pre5.0"`
```

The native go tooling already has support for Doc generation based on comments:
```go
	// Name of the service. This needs to be changed when using more than one app provider.
        // Each app provider configured needs to be identified by a unique service name.
        // Possible examples are:
        // * 'app-provider-collabora',
        // * 'app-provider-onlyoffice'
        // * 'app-provider-office365'
	Name string `yaml:"name" env:"APP_PROVIDER_SERVICE_NAME" introductionVersion:"pre5.0"`
```

This would not only allow using markdown, but also the [`Deprecated`](https://go.dev/wiki/Deprecated) convention understood by IDEs and other tooling.

Further reading: https://medium.com/@helbingxxx/how-to-write-go-doc-comments-421e0ca85996

**Comment by kobergj** [comment]
The question is if we can parse this style of comments for the docs and devdocs. I'm not really sure we want to reimplement all the magic that `godoc` does to doc strings. Maybe we would need to run it through godoc generation and then integrate this in the docs somehow?

@mmattel @dragonchaser any opinion on this?

**Comment by micbar** [comment]
I am also really questioning the effort vs value in this ticket.

I vote for keeping the process as it is.

**Comment by dragonchaser** [comment]
I agree with @micbar. Changing this would result in a complete rewrite of the tooling in `docs/helpers`. Too much effort for too little value IMHO.

**Comment by mmattel** [comment]
From the docs pov there are some points:
* Any change will more or less not affect the output printed in the admin docs - if covered by the docs helper. Rendering would be the same.
* Code readability would definitely increase for the sake of the work to make it happen.
* The effort to change that is two fold:
  - You have to touch ALL envvars.
  - You have to adapt the docs helpers code.

Saying that, this is a dev and no admin docs issue.

**Comment by dragonchaser** [comment]
@mmattel not correct, the descriptions for the admin doc is extracted from the variable annotation, if you change that the tooling has to change as mentioned above.

**Comment by micbar** [comment]
Closing. The current approach is working.
